google-cloud-storage/src/test/java/com/google/cloud/storage/it/ITStorageTest.java

```diff
@@ -1448,24 +1490,41 @@ public void testReadCompressedBlob() throws IOException {
 
   @Test
   public void testBucketPolicy() {
+    testBucketPolicyRequesterPays(true);
+    testBucketPolicyRequesterPays(false);
+  }
+
+  private void testBucketPolicyRequesterPays(boolean requesterPays) {
+    if (requesterPays) {
+      Bucket remoteBucket = storage.get(BUCKET, Storage.BucketGetOption.fields(BucketField.ID));
+      assertNull(remoteBucket.requesterPays());
+      remoteBucket = remoteBucket.toBuilder().setRequesterPays(true).build();
+      Bucket updatedBucket = storage.update(remoteBucket);
+      assertTrue(updatedBucket.requesterPays());
+    }
+
     String projectId = remoteStorageHelper.getOptions().getProjectId();
+
+    Storage.BucketSourceOption[] bucketOptions = requesterPays
+        ? new Storage.BucketSourceOption[] {Storage.BucketSourceOption.userProject(projectId)}
+        : new Storage.BucketSourceOption[] {};
     Identity projectOwner = Identity.projectOwner(projectId);
     Identity projectEditor = Identity.projectEditor(projectId);
     Identity projectViewer = Identity.projectViewer(projectId);
     Map<com.google.cloud.Role, Set<Identity>> bindingsWithoutPublicRead =
         ImmutableMap.of(
             StorageRoles.legacyBucketOwner(),
-            (Set<Identity>) newHashSet(projectOwner, projectEditor),
-            StorageRoles.legacyBucketReader(), newHashSet(projectViewer));
+            (Set<Identity>) Sets.newHashSet(projectOwner, projectEditor),
```

Comment:
Please, since you already touching this, replace `Sets.newHashSet()` with just `new HashSet()` call, as it is mentioned in `Sets.newHashSet()` guava documentation:
```java
  /**
   * <p><b>Note for Java 7 and later:</b> this method is now unnecessary and should be treated as
   * deprecated. Instead, use the {@code HashSet} constructor directly, taking advantage of the new
   * <a href="http://goo.gl/iz2Wi">"diamond" syntax</a>.
  */
```

Reply:
Done.

Reply:
Done.